quickwit/quickwit-indexing/src/merge_policy/mod.rs

```diff
@@ -160,4 +211,197 @@ pub mod tests {
             })
             .collect()
     }
+
+    // Creates a checksum for a given merge operation.
+    // This does not take in account the merge split id,
+    // and is split order independent.
+    fn compute_checksum_op(op: &MergeOperation) -> u64 {
+        let mut checksum = 0u64;
+        for split in op.splits_as_slice() {
+            let mut hasher = DefaultHasher::default();
+            hasher.write(split.split_id.as_bytes());
+            checksum ^= hasher.finish();
+        }
+        checksum
+    }
+
+    // Creates a checksum for a set of operations.
+    // This checksum does not depend on the order of the merrge operations,
+    // nor the merge split ids.
+    fn compute_checksum_ops(ops: &[MergeOperation]) -> u64 {
+        let mut checksum = 0u64;
+        for op in ops {
+            let op_checksum = compute_checksum_op(op);
+            let mut hasher = DefaultHasher::default();
+            hasher.write_u64(op_checksum);
+            checksum ^= hasher.finish();
+        }
+        checksum
+    }
+
+    fn compare_merge_operations(left_ops: &[MergeOperation], right_ops: &[MergeOperation]) -> bool {
+        compute_checksum_ops(left_ops) == compute_checksum_ops(right_ops)
+    }
+
+    pub(crate) fn proptest_merge_policy(merge_policy: &dyn MergePolicy) {
+        proptest!(|(mut splits in prop::collection::vec(split_strategy(), 0..100))| {
+            let mut cloned_splits = splits.clone();
+            cloned_splits.shuffle(&mut rand::thread_rng());
+
+            let original_num_splits = splits.len();
+
+            let mut operations: Vec<MergeOperation> = merge_policy.operations(&mut splits);
+            let operations_after_shuffle = merge_policy.operations(&mut cloned_splits);
+            assert!(compare_merge_operations(&operations[..],
+                &operations_after_shuffle[..]),
+                "Merge policy result should be independent from the original order.");
+
+            let num_splits_in_merge: usize = operations.iter().map(|op| op.splits_as_slice().len()).sum();
+
+            assert_eq!(
+                num_splits_in_merge + splits.len(), original_num_splits,
+                "Splits should not be lost."
+            );
+
+            // assert!(
+            //     merge_policy.operations(&mut splits).is_empty(),
+            //     "Merge policy are expected to return all available merge operations."
+            // );
+
```

Comment:
```suggestion
```

Reply:
The stable log merge policy does not observe this actually.